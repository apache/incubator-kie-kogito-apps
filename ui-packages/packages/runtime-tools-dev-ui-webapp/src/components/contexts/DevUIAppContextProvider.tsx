import React from 'react';
import { User } from '@kogito-apps/consoles-common/dist/environment/auth';
import RuntimeToolsDevUIAppContext, {
  DevUIAppContextImpl
} from './DevUIAppContext';
import { CustomLabels } from '../../api/CustomLabels';
import { DiagramPreviewSize } from '@kogito-apps/process-details/dist/api';

interface IOwnProps {
  users: User[];
  devUIUrl: string;
  openApiPath: string;
  isProcessEnabled: boolean;
  isTracingEnabled: boolean;
  availablePages: string[];
  customLabels: CustomLabels;
  omittedProcessTimelineEvents: string[];
  diagramPreviewSize: DiagramPreviewSize;
  isStunnerEnabled: boolean;
}

const DevUIAppContextProvider: React.FC<IOwnProps> = ({
  users,
  devUIUrl,
  openApiPath,
  isProcessEnabled,
  isTracingEnabled,
  availablePages,
  customLabels,
  omittedProcessTimelineEvents,
  diagramPreviewSize,
  isStunnerEnabled,
  children
}) => {
  return (
    <RuntimeToolsDevUIAppContext.Provider
      value={
        new DevUIAppContextImpl({
          users,
          devUIUrl,
          openApiPath,
          isProcessEnabled,
          isTracingEnabled,
          availablePages,
          customLabels,
          omittedProcessTimelineEvents,
          diagramPreviewSize,
          isStunnerEnabled
        })
      }
    >
      {children}
    </RuntimeToolsDevUIAppContext.Provider>
  );
};

export default DevUIAppContextProvider;
