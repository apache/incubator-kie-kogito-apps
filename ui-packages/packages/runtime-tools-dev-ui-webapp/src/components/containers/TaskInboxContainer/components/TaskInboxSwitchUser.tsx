import React, { useState } from 'react';
import {
  Dropdown,
  DropdownToggle,
  DropdownItem,
  Tooltip
} from '@patternfly/react-core';
import { ThIcon } from '@patternfly/react-icons';
import { useDevUIAppContext } from '../../../contexts/DevUIAppContext';

const TaskInboxSwitchUser = () => {
  const [isOpen, setIsOpen] = useState(false);
  const appContext = useDevUIAppContext();
  const allUsers = appContext.getAllUsers();

  const onSelect = (event): void => {
    const selectedUser = event.target.innerHTML;
    appContext.switchUser(selectedUser);

    setIsOpen(!isOpen);
  };

  const dropdownItems = (): JSX.Element[] => {
    const userIds = [];
    allUsers.map(user => {
      userIds.push(<DropdownItem key={user.id}>{user.id}</DropdownItem>);
    });
    return userIds;
  };

  const onToggle = (isOpen): void => {
    setIsOpen(isOpen);
  };

  return (
    <Dropdown
      onSelect={onSelect}
      toggle={
        <DropdownToggle
          toggleIndicator={null}
          onToggle={onToggle}
          aria-label="Applications"
          id="toggle-id-7"
        >
          <Tooltip position="top" content={<div>Switch user</div>}>
            <ThIcon />
          </Tooltip>
        </DropdownToggle>
      }
      isOpen={isOpen}
      isPlain
      dropdownItems={dropdownItems()}
    />
  );
};

export default TaskInboxSwitchUser;
