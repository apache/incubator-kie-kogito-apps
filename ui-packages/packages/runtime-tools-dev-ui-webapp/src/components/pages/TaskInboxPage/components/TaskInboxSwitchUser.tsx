import React, { useState } from 'react';
import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
import { UserIcon, CaretDownIcon } from '@patternfly/react-icons';
import { useDevUIAppContext } from '../../../contexts/DevUIAppContext';
import '../../../styles.css';

interface IOwnProps {
  user: string;
}

const TaskInboxSwitchUser: React.FC<IOwnProps> = ({ user }) => {
  const appContext = useDevUIAppContext();
  const [isOpen, setIsOpen] = useState(false);
  const [currentUser, setCurrentUser] = useState(user);
  const allUsers = appContext.getAllUsers();

  const onSelect = (event): void => {
    const selectedUser = event.target.innerHTML;
    appContext.switchUser(selectedUser);
    setCurrentUser(selectedUser);
    setIsOpen(!isOpen);
  };

  const dropdownItems = (): JSX.Element[] => {
    const userIds = [];
    allUsers.map(user => {
      userIds.push(<DropdownItem key={user.id}>{user.id}</DropdownItem>);
    });
    return userIds;
  };

  const onToggle = (isOpen): void => {
    setIsOpen(isOpen);
  };

  return (
    <Dropdown
      onSelect={onSelect}
      toggle={
        <DropdownToggle
          onToggle={onToggle}
          aria-label="Applications"
          id="toggle-id-7"
          toggleIndicator={CaretDownIcon}
          icon={<UserIcon />}
        >
          {currentUser}
        </DropdownToggle>
      }
      isOpen={isOpen}
      isPlain
      dropdownItems={dropdownItems()}
      className="DevUI-switchUser-dropdown-styling"
    />
  );
};

export default TaskInboxSwitchUser;
