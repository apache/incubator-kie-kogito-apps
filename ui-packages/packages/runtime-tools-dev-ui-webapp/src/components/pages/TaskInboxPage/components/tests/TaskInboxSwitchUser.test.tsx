/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import TaskInboxSwitchUser from '../TaskInboxSwitchUser';
import DevUIAppContextProvider from '../../../../contexts/DevUIAppContextProvider';
import { act } from 'react-dom/test-utils';

describe('TaskInboxSwitchUser tests', () => {
  it('Snapshot test with default props', () => {
    const wrapper = mount(
      <DevUIAppContextProvider users={[{ id: 'John snow', groups: ['admin'] }]}>
        <TaskInboxSwitchUser user="John" />
      </DevUIAppContextProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Trigger onSelect event', () => {
    const wrapper = mount(
      <DevUIAppContextProvider users={[{ id: 'John snow', groups: ['admin'] }]}>
        <TaskInboxSwitchUser user="John" />
      </DevUIAppContextProvider>
    );

    const event: any = { target: { innerHTML: 'admin' } };
    act(() => {
      wrapper
        .find('Dropdown')
        .props()
        ['onSelect'](event);
    });
    wrapper.update();
    expect(
      wrapper
        .find('Text')
        .find('p')
        .text()
    ).toEqual('admin');
  });

  it('Trigger toggle event', () => {
    const wrapper = mount(
      <DevUIAppContextProvider users={[{ id: 'John snow', groups: ['admin'] }]}>
        <TaskInboxSwitchUser user="John" />
      </DevUIAppContextProvider>
    );

    act(() => {
      wrapper
        .find('DropdownToggle')
        .props()
        ['onToggle']();
    });
  });
});
