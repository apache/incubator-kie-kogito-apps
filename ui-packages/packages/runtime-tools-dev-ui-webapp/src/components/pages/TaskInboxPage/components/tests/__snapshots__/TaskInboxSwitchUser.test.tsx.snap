// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInboxSwitchUser tests Snapshot test with default props 1`] = `
<DevUIAppContextProvider
  users={
    Array [
      Object {
        "groups": Array [
          "admin",
        ],
        "id": "John snow",
      },
    ]
  }
>
  <TaskInboxSwitchUser
    user="John"
  >
    <Dropdown
      className="DevUI-switchUser-dropdown-styling"
      dropdownItems={
        Array [
          <DropdownItem>
            John snow
          </DropdownItem>,
        ]
      }
      isOpen={false}
      isPlain={true}
      onSelect={[Function]}
      toggle={
        <DropdownToggle
          aria-label="Applications"
          icon={
            <UserIcon
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          }
          id="toggle-id-7"
          onToggle={[Function]}
          toggleIndicator={[Function]}
        >
          John
        </DropdownToggle>
      }
    >
      <DropdownWithContext
        autoFocus={true}
        className="DevUI-switchUser-dropdown-styling"
        direction="down"
        dropdownItems={
          Array [
            <DropdownItem>
              John snow
            </DropdownItem>,
          ]
        }
        isGrouped={false}
        isOpen={false}
        isPlain={true}
        menuAppendTo="inline"
        onSelect={[Function]}
        position="left"
        toggle={
          <DropdownToggle
            aria-label="Applications"
            icon={
              <UserIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            id="toggle-id-7"
            onToggle={[Function]}
            toggleIndicator={[Function]}
          >
            John
          </DropdownToggle>
        }
      >
        <div
          className="pf-c-dropdown DevUI-switchUser-dropdown-styling"
          data-ouia-component-id="OUIA-Generated-Dropdown-1"
          data-ouia-component-type="PF4/Dropdown"
          data-ouia-safe={true}
        >
          <DropdownToggle
            aria-haspopup={true}
            aria-label="Applications"
            getMenuRef={[Function]}
            icon={
              <UserIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            id="toggle-id-7"
            isOpen={false}
            isPlain={true}
            key=".0"
            onEnter={[Function]}
            onToggle={[Function]}
            parentRef={
              Object {
                "current": <div
                  class="pf-c-dropdown DevUI-switchUser-dropdown-styling"
                  data-ouia-component-id="OUIA-Generated-Dropdown-1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Applications"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe="true"
                    id="toggle-id-7"
                    type="button"
                  >
                    <span
                      class="pf-c-dropdown__toggle-image"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="pf-c-dropdown__toggle-text"
                    >
                      John
                    </span>
                    <span
                      class="pf-c-dropdown__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>,
              }
            }
            toggleIndicator={[Function]}
          >
            <Toggle
              aria-haspopup={true}
              aria-label="Applications"
              bubbleEvent={false}
              className=""
              data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
              data-ouia-component-type="PF4/DropdownToggle"
              data-ouia-safe={true}
              getMenuRef={[Function]}
              id="toggle-id-7"
              isActive={false}
              isDisabled={false}
              isOpen={false}
              isPlain={true}
              isPrimary={false}
              isSplitButton={false}
              onEnter={[Function]}
              onToggle={[Function]}
              parentRef={
                Object {
                  "current": <div
                    class="pf-c-dropdown DevUI-switchUser-dropdown-styling"
                    data-ouia-component-id="OUIA-Generated-Dropdown-1"
                    data-ouia-component-type="PF4/Dropdown"
                    data-ouia-safe="true"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-label="Applications"
                      class="pf-c-dropdown__toggle pf-m-plain"
                      data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                      data-ouia-component-type="PF4/DropdownToggle"
                      data-ouia-safe="true"
                      id="toggle-id-7"
                      type="button"
                    >
                      <span
                        class="pf-c-dropdown__toggle-image"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                          />
                        </svg>
                      </span>
                      <span
                        class="pf-c-dropdown__toggle-text"
                      >
                        John
                      </span>
                      <span
                        class="pf-c-dropdown__toggle-icon"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 320 512"
                          width="1em"
                        >
                          <path
                            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>,
                }
              }
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Applications"
                className="pf-c-dropdown__toggle pf-m-plain"
                data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                data-ouia-component-type="PF4/DropdownToggle"
                data-ouia-safe={true}
                disabled={false}
                id="toggle-id-7"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <span
                  className="pf-c-dropdown__toggle-image"
                >
                  <UserIcon
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                      />
                    </svg>
                  </UserIcon>
                </span>
                <span
                  className="pf-c-dropdown__toggle-text"
                >
                  John
                </span>
                <span
                  className="pf-c-dropdown__toggle-icon"
                >
                  <CaretDownIcon
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </CaretDownIcon>
                </span>
              </button>
            </Toggle>
          </DropdownToggle>
        </div>
      </DropdownWithContext>
    </Dropdown>
  </TaskInboxSwitchUser>
</DevUIAppContextProvider>
`;
